examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -327,17 +346,38 @@ func copyableType(t *types.Type) bool {
 	if ttag != nil && ttag.value == "false" {
 		return false
 	}
-	// TODO: Consider generating functions for other kinds too.
-	if t.Kind != types.Struct {
+
+	// Filter out private types.
+	if namer.IsPrivateGoName(t.Name.Name) {
 		return false
 	}
-	// Also, filter out private types.
-	if namer.IsPrivateGoName(t.Name.Name) {
+
+	if t.Kind == types.Alias {
+		// if the underlying built-in is not deepcopy-able, deepcopy is opt-in through definition of custom methods.
+		if hasDeepCopyMethod(t) || hasDeepCopyIntoMethod(t) {
+			return true
+		} else {
+			return t.Underlying.Kind != types.Builtin || copyableType(t.Underlying)
```

Comment:
won't this catch Map and Slice when it didn't used to?

Reply:
Yes, it does. It's intentional, compare https://goplay.space/#KWig91IjuQv

Will add a comment.